README.md

```diff
@@ -167,3 +219,42 @@ process.on('SIGINT', () => {
   finalize().catch(console.error);
 });
 ```
+
+### Binding resources
+
+You can use `bindResource` instead of normal `bind` which automatically registers `finalize` method as the finalizer.
+
+```typescript
+class Finalizable {
+    public async finalize() {
+        console.log('cleanup');
+    }
+}
+
+// These two designs are equivalent.
+Design.bind('finalizable', injectClass(Finalizable, []), resource => resource.finalize());
+Design.bindResource('finalizable', injectClass(Finalizable, []));
+```
+
+The combination of `inject` and `bindResource` lets you easily bind your own resource class which needs initialization and finalization to a design.
+
+```typescript
```

Comment:
I think reversing those two example helps readers understanding.

Because `Design.bindResource('resource', inject(Resource.initialize, ['config']))` is more general way than the specialized form `Design.bindResource('finalizable', injectClass(Finalizable, []));` .

Reply:
Actually I noticed we don't even need to use `injectXx` for this example, so I'll use `() => new Finalize()` here.